dht.go

```diff
@@ -114,15 +119,16 @@ func New(ctx context.Context, h host.Host, options ...opts.Option) (*IpfsDHT, er
 	dht.enableProviders = cfg.EnableProviders
 	dht.enableValues = cfg.EnableValues
 
+	subnot := (*subscriberNotifee)(dht)
+
 	// register for network notifs.
-	dht.host.Network().Notify((*netNotifiee)(dht))
+	dht.host.Network().Notify(subnot)
 
-	dht.proc = goprocessctx.WithContextAndTeardown(ctx, func() error {
-		// remove ourselves from network notifs.
-		dht.host.Network().StopNotify((*netNotifiee)(dht))
-		return nil
-	})
+	dht.proc = goprocessctx.WithContext(ctx)
 
+	// register for network notifs.
+	dht.proc.Go((*subscriberNotifee)(dht).subscribe)
```

Comment:
```suggestion
	dht.proc.Go(subnot.subscribe)
```

Reply:
Going to leave as is and just remove the subnot variable declaration. Also, going to keep the Notifier in the subscription goroutine instead of external (i.e. solving the duplication mentioned at https://github.com/libp2p/go-libp2p-kad-dht/pull/472#discussion_r387423917)